### PR TITLE
External icon libraries & token generation

### PR DESCRIPTION
**Pull request by CrookedGrin**
This update does two things:

1. Wraps up the external icon library registration into a single exported method that can be called as-is.This will allow the Rails app, the docs site, and CodePen to all just call `registerExternalLibraries()` instead of duplicating the registration logic for each library like they do currently.

2. Creates a single source-of-truth for color and font tokens in JSON and then generates all the CSS tokens via automation. In addition to cleaning things up, this will allow us to import that same JSON into shared-ui and use it to generate the tailwind config, and thus avoid having our color tokens defined in multiple places. The tokens are exported and available via CDN or through the package itself via Yarn/NPM.

### CODE REVIEW

File: docs/assets/plugins/code-block/code-block.js [hunk 1]
```diff
@@ -343,12 +343,13 @@
       form.method = 'POST';
       form.target = '_blank';
 
+      // `<script type="module" src="https://cdn.jsdelivr.net/npm/@teamshares/shoelace@${version}/dist/shoelace.js"></script>\n` +
```
Comment (kdonovan) [inline docs/assets/plugins/code-block/code-block.js]: Do we still want this comment as a reference?

File: scripts/make-styles.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { mkdirSync } from 'fs';
 import { globbySync } from 'globby';
 import path from 'path';
 import prettier from 'prettier';
-import stripComments from 'strip-css-comments';
+// import stripComments from 'strip-css-comments';
```
Comment (kdonovan) [inline scripts/make-styles.js]: ??

File: scripts/make-tokens.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
+/* Read from the JSON source of truth and write a css file using the template */
+
+import chalk from 'chalk';
+import fs from 'fs';
+import path from 'path';
+import prettier from 'prettier';
+import tokens from './../src/styles/tokens.json' assert { type: 'json' };
```
Comment (kdonovan) [inline scripts/make-tokens.js:7]: TIL `import` + assert`

File: src/styles/exports/overrides.css [hunk 4]
```diff
@@ -580,12 +582,12 @@ sl-dialog.small::part(body) {
     --header-spacing: 32px 0 16px 24px;
   }
 
-  sl-dialog.has-icon > div {
+  sl-dialog.has-icon>div {
```
Comment (kdonovan) [inline src/styles/exports/overrides.css]: The syntax highlighting implies the lack of spaces around `>` is fine, just calling it out to be sure :)

**Review by kdonovan — APPROVED** [review]
Thanks, Daross -- awesome work automating the generation/letting us centralize configs! 🚗

File: src/styles/exports/overrides.css (hunk 4 above)
Comment (CrookedGrin) [reply src/styles/exports/overrides.css]: I had initially added these files to .prettierignore, but trying out putting them back in

File: scripts/make-tokens.js (hunk 3 above)
Comment (CrookedGrin) [reply scripts/make-tokens.js:7]: Yeah, that's a new / experimental Node thing. You get a warning in the console, but it works.

File: src/styles/exports/overrides.css (hunk 4 above)
Comment (CrookedGrin) [reply src/styles/exports/overrides.css]: I ended up being able to just ignore one of the CSS files (the generated one), which for whatever reason fails the prettier check on CI, but not locally.

**Pull request merged**